docs/clients/grpc/observability.md

```diff
@@ -0,0 +1,106 @@
+# Observability
+
+The EventStoreDB gRPC clients are designed with observability in mind, offering support for OpenTelemetry. This integration provides a set of distributed traces, enabling developers to gain deeper insights into their system.
+
+::: warning
+Currently not all clients have support for observability using OpenTelemetry.
+:::
+
+::: note
+Only the append and subscribe (Catchup and Persistent) operations are instrumented.
+:::
+
+## Traces
+
+Traces provides a clear picture of how operations are carried out in a distributed system, making it easier to maintain and enhance the system over time. Traces from the clients can be exported to any compatible collector that supports the OpenTelemetry protocol (OTLP).
+
+For guidance on setting up and utilizing tracing, refer to the [OpenTelemetry](https://opentelemetry.io/) documentation.
+
+### Install Required Packages
+
+:::: code-group
+::: code-group-item JavaScript
+
+```
+# Yarn
+$ yarn add @eventstore/opentelemetry
+
+# NPM
+$ npm install --save @eventstore/opentelemetry
+```
+
+:::
+::: code-group-item TypeScript
+
+```
+# TypeScript Declarations are included in the package.
+
+# Yarn
+$ yarn add @eventstore/opentelemetry
+
+# NPM
+$ npm install --save @eventstore/opentelemetry
+```
+
+:::
+::: code-group-item Java
+
+```
+OpenTelemetry is supported out of the box in the Java client.
+```
+
+:::
+::: code-group-item C#
+
+```bash
+dotnet add package EventStore.Client.Extensions.OpenTelemetry
+```
+
+:::
+::::
+
+You can click on the links below to view the full code for each client:
```

Comment:
Because these samples are the full samples from installation to registration to exporters, we should probably move this list up into the intro. And the sentence could read something like: "Follow the links below to view sample code from installation to configuration for each client." We may also want to include if this "full code" can just be copied and pasted into a testing or production env.

Reply:
I initially aimed to have a button on top of each snippet that redirects users to the full code for context. However, I didn't find an easy way to do this. From a user's perspective, we only care about knowing how to install the required package and register the instrumentation. It's then up to them to decide how they want to export the traces, whether it's to the console, Jaeger, etc.

Initially, I didn't want to include the "With Exporters" section because I assumed developers already know how to do this or can read the OpenTelemetry documentation. However, I wanted to display it step by step so that a new person can follow along and understand what each part means.

Another approach would be to include everything in a single snippet under a single header (Traces), but if we later decide to add metrics, there will be a lot of redundant code.

To address this, I will have traces in their own dedicated section.

Reply:
Makes sense. I appreciate the exporters section because we don't want to assume anyone's starting level. If you think these links will fit better under one of the subsections when we add them later, we can move them.